src/ticks.js

```diff
@@ -0,0 +1,75 @@
+import {bisector, tickStep} from "d3-array";
+
+import millisecond from "./millisecond";
+import second from "./second";
+import minute from "./minute";
+import hour from "./hour";
+import day from "./day";
+import {default as week} from "./week";
+import month from "./month";
+import year from "./year";
+
+var millisecondsPerSecond = 1000,
+    millisecondsPerMinute = millisecondsPerSecond * 60,
+    millisecondsPerHour = millisecondsPerMinute * 60,
+    millisecondsPerDay = millisecondsPerHour * 24,
+    millisecondsPerWeek = millisecondsPerDay * 7,
+    millisecondsPerMonth = millisecondsPerDay * 30,
+    millisecondsPerYear = millisecondsPerDay * 365,
+    bisectTickIntervals = bisector(function(method) { return method[2]; }).right;
+
+export default function(start, stop, count) {
+  start = new Date(+start);
+  var info = tickInterval(start, stop, count),
+    step = 0,
+    nextDate = start,
+    range = [nextDate];
+  while (nextDate = info.interval.offset(start, step += info.step), nextDate <= +stop) range.push(nextDate);
```

Comment:
This follows d3-array.ticks more than interval.range in that it starts from the exact start date and then steps until before the stop date. Again the problem of not being clear on the use case is I'm not sure if nice endpoints is preferable here.